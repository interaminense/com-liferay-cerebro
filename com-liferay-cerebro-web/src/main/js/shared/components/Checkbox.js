import Component, {Config} from 'metal-jsx';
import getCN from 'classnames';

const SIZES = ['sm', 'default', 'md', 'lg'];

class Checkbox extends Component {
	syncIndeterminate(newVal, prevVal) {
		if (newVal !== prevVal) {
			this.refs.checkbox.indeterminate = newVal;
		}
	}

	render() {
		const {checked, disabled, displayInline, label, name} = this.props;

		const classes = getCN('custom-control', 'custom-checkbox', {
			['custom-control-inline']: displayInline
		});

		return (
			<div class={classes}>
				<label {...this.otherProps()}>
					<input
						checked={checked}
						class="custom-control-input"
						disabled={disabled}
						name={name}
						ref="checkbox"
						type="checkbox"
						value={checked}
					/>

					<span class="custom-control-indicator" />

					<span class="custom-control-description">{label}</span>
				</label>
			</div>
		);
	}
}

Checkbox.PROPS = {
	checked: Config.bool(),
	disabled: Config.bool(),
	displayInline: Config.bool(),
	elementClasses: Config.string(),
	indeterminate: Config.bool(),
	label: Config.string(),
	name: Config.string()
};

Checkbox.SIZES = SIZES;

export default Checkbox;