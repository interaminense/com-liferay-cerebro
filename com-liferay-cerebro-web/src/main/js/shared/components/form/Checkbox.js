import autobind from 'autobind-decorator';
import Checkbox from '../Checkbox';
import FormComponent, {Config} from './Component';

class FormCheckbox extends FormComponent {
	created() {
		this.state.value = this.props.initialValue;
	}

	@autobind
	handleChange() {
		const {onChange} = this.props;

		const newVal = !this.state.value;

		this.state.value = newVal;

		if (onChange) {
			onChange(newVal);
		}
	}

	syncInitialValue(newVal) {
		this.state.value = newVal;
	}

	render() {
		return (
			<Checkbox
				{...this.otherProps()}
				name={this.props.name}
				onChange={this.handleChange}
			/>
		);
	}
}

FormCheckbox.PROPS = {
	initialValue: Config.bool().value(false),
	name: Config.string().required(),
	onChange: Config.func()
};

FormCheckbox.STATE = {
	value: Config.value(false)
};

export default FormCheckbox;