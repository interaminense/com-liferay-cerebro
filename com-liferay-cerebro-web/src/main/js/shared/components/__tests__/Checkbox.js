import Checkbox from '../Checkbox';

describe('Checkbox', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new Checkbox();

		expect(component).toMatchSnapshot();
	});
});