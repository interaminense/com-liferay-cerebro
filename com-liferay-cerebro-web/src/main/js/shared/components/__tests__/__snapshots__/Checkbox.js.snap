// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render 1`] = `
<div class="custom-control custom-checkbox">
  <label>
    <input class="custom-control-input" ref="checkbox" type="checkbox">
    <span class="custom-control-indicator"></span>
    <span class="custom-control-description"></span>
  </label>
</div>
`;
